docs/transaction.md

```diff
@@ -0,0 +1,261 @@
+# Life of a transaction
+
+This document describes how Dragonfly transactions provide atomicity and serializability for its multi-key and multi-command operations.
+
+## Definitions
+
+### Serializability
+
+Serializability is an isolation level for database transactions. Serializability describes multiple transactions, where a transaction is usually composed of multiple operations on multiple objects.
+
+Database can executed transactions in parallel (and the operations in parallel). Serializability guarantees the result is the same with, as if the transactions were executed one by one. i.e. to behave like executed in a serial order.
+
+Serializability doesn’t guarantee the resulting serial order respects recency. I.e. the serial order can be different from the order in which transactions were actually executed. E.g. Tx1 begins earlier than Tx2, but the result behaves as if Tx2 executed before Tx1. That is also to say, to satisfy the same Serializability, there can be more than one possible execution schedulings.
+
+### Linearizability
+
+Linearizability (also known as atomic consistency, strong consistency, immediate consistency) describes reads and writes on a single object (stores a single value). it doesn’t involve multiple objects. It doesn’t involve “transaction”, which groups multiple objects. It treats each operation as an atom, i.e. to take effect in a single time point, rather than a timespan.
+
+Note that simple, single-key operations in Dragonfly are already linearizable. Indeed, in a shared-nothing architecture each shard-thread performs operations on its keys sequentially.
+The complexity rises when we need to provide strict-serializability (aka serializability and linearizability) for multiple keys.
+
+## Transactions high level overview
+Transactions in Dragonfly are orchestrated by an abstract entity, called coordinator.
+In reality, a client connection instance takes on itself the role of a coordinator: it coordinates transactions every time it executes a redis or memcached command. The algorithm behind Dragonfly transactions is based on the [VLL paper](https://www.cs.umd.edu/~abadi/papers/vldbj-vll.pdf).
```

Comment:
The coordinator is handling a single command every time till it finishes right ? 
I would not say executes a command because it mixes with the execute phase below, use another word.
What is to coordinate a transaction?

Reply:
a client connection instance takes on itself the role of a coordinator.The transaction to 2 phases schedule and execution, which might consist of one or more steps. The coordinator is responsible to orchestrate this 2 phases and the steps inside.

Reply:
rewritten differently.